File: sample/src/test/java/vn/tiki/sample/util/EmailValidatorTest.java
package vn.tiki.sample.util;

import org.junit.Before;
import org.junit.Test;

import static com.google.common.truth.Truth.assertThat;

public class EmailValidatorTest {
  private EmailValidator tested;

  @Before
  public void setUp() throws Exception {
    tested = new EmailValidator();
  }

  @Test
  public void testValidate_invalidEmail() throws Exception {
    assertThat(tested.validate("foo")).isFalse();
    assertThat(tested.validate("foo@")).isFalse();
    assertThat(tested.validate("foo@gmail")).isFalse();
  }

  @Test
  public void testValidate_validEmail() throws Exception {
    assertThat(tested.validate("foo@gmail.com")).isTrue();
  }
}

File: sample/src/test/java/vn/tiki/sample/util/EmailValidatorTest.kt
package vn.tiki.sample.util

import com.google.common.truth.Truth.assertThat
import org.junit.*

class EmailValidatorTest {
  private lateinit var tested: EmailValidator

  @Before
  fun setUp() {
    tested = EmailValidator()
  }

  @Test
  fun testValidateInvalidEmail() {
    assertThat(tested.validate("foo")).isFalse()
    assertThat(tested.validate("foo@")).isFalse()
    assertThat(tested.validate("foo@gmail")).isFalse()
  }

  @Test
  fun testValidateValidEmail() {
    assertThat(tested.validate("foo@gmail.com")).isTrue()
  }
}
